addons/project/views/project_sharing_project_task_views.xml

```diff
@@ -192,7 +192,7 @@
                                     <field name="allow_milestones" column_invisible="True"/>
                                     <field name="milestone_id"
                                         optional="hide"
-                                        context="{'default_project_id': project_id}"
+                                        context="{'default_project_id': project_id, 'is_sharing_milestone_field': True}"
```

Comment:
Instead of doing that, we should probably add the same options than the one in line 161. 🤷🏼‍♂️